### ISSUE TITLE
query results are not same when turn on pipeline

### ISSUE THREAD

**Issue by colorfulu** [StarRocks/starrocks#2640]
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS w0;
CREATE DATABASE w0;
USE w0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 TINYINT NOT NULL ,c_0_1 DECIMAL128(35, 14)  NULL ,c_0_2 DECIMAL32(3, 3)  NOT NULL ,c_0_3 DECIMAL(7, 0)  NULL ,c_0_4 DATETIME NULL ,c_0_5 DECIMAL(17, 8)  NOT NULL ,c_0_6 DECIMAL32(3, 0)  NULL ,c_0_7 BOOLEAN NULL ,c_0_8 DATETIME NULL ,c_0_9 DECIMAL(18, 1)  NULL ,c_0_10 BOOLEAN NOT NULL ,c_0_11 DECIMAL128(27, 14)  NULL ,c_0_12 DECIMAL64(1, 1)  NOT NULL ,c_0_13 DECIMAL32(2, 2)  NULL ,c_0_14 DECIMAL32(6, 5)  NULL ,c_0_15 DECIMAL128(10, 8)  NULL ,c_0_16 DECIMAL(14, 9)  NULL ) UNIQUE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8) DISTRIBUTED BY HASH (c_0_0,c_0_2,c_0_4,c_0_7) properties("replication_num"="1") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATE NULL ,c_1_1 TINYINT NULL ,c_1_2 DECIMAL32(7, 0)  NULL ,c_1_3 BOOLEAN NULL ,c_1_4 DECIMAL128(37, 32)  REPLACE NOT NULL ,c_1_5 HLL HLL_UNION NULL ) AGGREGATE KEY (c_1_0,c_1_1,c_1_2,c_1_3) DISTRIBUTED BY HASH (c_1_1,c_1_2) properties("replication_num"="1") ;
CREATE TABLE t2 (c_2_0 DECIMAL128(8, 4)  NOT NULL ) DUPLICATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) properties("replication_num"="1") ;
CREATE TABLE t3 (c_3_0 DECIMAL64(6, 6)  NOT NULL ,c_3_1 BITMAP BITMAP_UNION NULL ,c_3_2 DATE REPLACE NULL ,c_3_3 HLL HLL_UNION NULL ,c_3_4 DECIMAL128(37, 11)  REPLACE_IF_NOT_NULL NULL ,c_3_5 INT SUM NULL ,c_3_6 BOOLEAN REPLACE NULL ,c_3_7 CHAR(1) MAX NULL ,c_3_8 DATE MIN NULL ,c_3_9 DOUBLE SUM NOT NULL ,c_3_10 BOOLEAN REPLACE NOT NULL ,c_3_11 BOOLEAN REPLACE NULL ,c_3_12 BITMAP BITMAP_UNION NOT NULL ) AGGREGATE KEY (c_3_0) DISTRIBUTED BY HASH (c_3_0) properties("replication_num"="1") ;
INSERT INTO t1 VALUES ("1970-01-01", 5, 5153189, false, 5.4371E-28, hll_hash("")), ("1970-01-01", 6, 3753993, false, 2.3136E-28, hll_hash("ihnF")), ("1970-01-01", 1, 7636290, NULL, 7.6740E-28, hll_hash("f"));
TRUNCATE TABLE t2;
INSERT INTO t1 VALUES ("1970-01-01", 1, 2544466, true, 8.0462E-28, hll_hash("e")), ("1970-01-01", 0, 8030790, true, 2.834E-29, hll_hash("WVyc")), ("1970-01-01", 6, 5744478, true, 4.6620E-28, hll_hash("")), ("1970-01-01", 1, 1110805, true, 5.6738E-28, hll_hash("qzMpcTuN"));
DELETE FROM t3 WHERE c_3_0=0.000003;
INSERT INTO t0 VALUES (7, 8968891.07584770955709, 0.008, 1190533, "1970-01-01 08:00:00", 2.15827532, 292, false, "1970-01-01 08:00:00", 2130246381270200.7, true, 0.01823138892109, 0.2, NULL, NULL, 1.6E-7, NULL);
INSERT INTO t1 VALUES (NULL, 5, 6405446, true, 3.2518E-28, hll_hash("OC3q")), ("1970-01-01", 7, 7373268, true, 6.7954E-28, hll_hash("8DZui"));
INSERT INTO t0 VALUES (6, 6480904.09514658818776, 0.007, 5738776, "1970-01-01 08:00:00", 6.29120661, 684, false, "1970-01-01 08:00:00", 9076329271638281.3, false, 0.00110969803435, 0.6, 0.01, 0.00007, 6.8E-7, 0.000026452), (7, 3664823.65239423600338, 0.003, 6159121, "1970-01-01 08:00:00", 8.43254981, 584, false, "1970-01-01 08:00:00", 2758030421697939.5, false, NULL, 0.4, 0.05, 0.00006, 8E-8, 0.000044181);
INSERT INTO t0 VALUES (6, 2620957.61390426703368, 0.006, NULL, "1970-01-01 08:00:00", 3.08382411, 90, true, "1970-01-01 08:00:00", 5661271411719451.1, true, 0.05754903177392, 0.2, 0.08, 0.00008, 8.3E-7, 0.000004460);
INSERT INTO t2 VALUES (0.1887);
INSERT INTO t0 VALUES (7, 2654179.16287350578622, 0.001, 6476006, "1970-01-01 08:00:00", 3.80694025, 434, true, "1970-01-01 08:00:00", 5193312474666668.3, true, 0.08316309766501, 0.4, 0.00, NULL, 7.3E-7, 0.000047506);
INSERT INTO t1 VALUES ("1970-01-01", 7, 9512477, true, 1.1614E-28, hll_hash("yFn")), ("1970-01-01", 2, 1468358, true, 4.2824E-28, hll_hash("sDhop"));
SELECT ((1598642779)/(531562539)) FROM t1,  (SELECT t0.c_0_0, t0.c_0_1, t0.c_0_2, t0.c_0_3, t0.c_0_4, t0.c_0_5, t0.c_0_6, t0.c_0_7, t0.c_0_8, t0.c_0_9, t0.c_0_10, t0.c_0_11, t0.c_0_12, t0.c_0_13, t0.c_0_14, t0.c_0_15, t0.c_0_16 FROM t0) subt0 LEFT ANTI JOIN (SELECT t2.c_2_0 FROM t2 WHERE false ORDER BY t2.c_2_0 ASC) subt2 ON subt0.c_0_12 = subt2.c_2_0 AND CASE  WHEN false THEN subt0.c_0_12 ELSE subt0.c_0_12 END != subt2.c_2_0 AND subt0.c_0_4 = subt2.c_2_0 AND subt0.c_0_4 != subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 WHERE false UNION ALL SELECT ((1598642779)/(531562539)) FROM t1,  (SELECT t0.c_0_0, t0.c_0_1, t0.c_0_2, t0.c_0_3, t0.c_0_4, t0.c_0_5, t0.c_0_6, t0.c_0_7, t0.c_0_8, t0.c_0_9, t0.c_0_10, t0.c_0_11, t0.c_0_12, t0.c_0_13, t0.c_0_14, t0.c_0_15, t0.c_0_16 FROM t0) subt0 LEFT ANTI JOIN (SELECT t2.c_2_0 FROM t2 WHERE false ORDER BY t2.c_2_0 ASC) subt2 ON subt0.c_0_12 = subt2.c_2_0 AND CASE  WHEN false THEN subt0.c_0_12 ELSE subt0.c_0_12 END != subt2.c_2_0 AND subt0.c_0_4 = subt2.c_2_0 AND subt0.c_0_4 != subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 WHERE (NOT (false)) UNION ALL SELECT ((1598642779)/(531562539)) FROM t1,  (SELECT t0.c_0_0, t0.c_0_1, t0.c_0_2, t0.c_0_3, t0.c_0_4, t0.c_0_5, t0.c_0_6, t0.c_0_7, t0.c_0_8, t0.c_0_9, t0.c_0_10, t0.c_0_11, t0.c_0_12, t0.c_0_13, t0.c_0_14, t0.c_0_15, t0.c_0_16 FROM t0) subt0 LEFT ANTI JOIN (SELECT t2.c_2_0 FROM t2 WHERE false ORDER BY t2.c_2_0 ASC) subt2 ON subt0.c_0_12 = subt2.c_2_0 AND CASE  WHEN false THEN subt0.c_0_12 ELSE subt0.c_0_12 END != subt2.c_2_0 AND subt0.c_0_4 = subt2.c_2_0 AND subt0.c_0_4 != subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 AND subt0.c_0_14 = subt2.c_2_0 WHERE ((false) IS NULL);
```
### Expected behavior (Required)
`52 rows in set`
### Real behavior (Required)
`49 rows in set`
`52 rows in set`
`55 rows in set`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+---------------------------------------+
| current_version()                     |
+---------------------------------------+
| PIPELINE_SQLANCER_MASTER_ASAN b857b8a |
```

**Comment by liuyehcf** [comment]
Can't reproduce in branch main

**Comment by liuyehcf** [comment]
Fix by #3776
